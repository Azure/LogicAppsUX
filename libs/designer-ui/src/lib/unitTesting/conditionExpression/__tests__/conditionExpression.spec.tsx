// biome-ignore lint/correctness/noUnusedImports: actually is used
import React from 'react';
import { ConditionExpression, type ConditionExpressionProps } from '../index';
import { initializeIcons } from '@fluentui/react';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import { describe, it, expect, beforeEach, vi, afterEach } from 'vitest';

describe('ui/unitTesting/conditionExpression/index', () => {
  let minimal: ConditionExpressionProps;
  let renderer: ReactShallowRenderer.ShallowRenderer;

  beforeEach(() => {
    minimal = {
      editorId: 'condition-expression-editor-id',
      labelId: 'condition-expression-label-id',
      initialValue: '@equals(1, 1)',
      filteredTokenGroup: [],
      tokenGroup: [],
      expressionGroup: [],
      getValueSegmentFromToken: vi.fn(),

      onChange: vi.fn(),
    };
    renderer = ReactShallowRenderer.createRenderer();
    initializeIcons();
  });

  afterEach(() => {
    renderer.unmount();
  });

  it('should render without assertions items and add button.', () => {
    const conditionExpression = renderer.render(<ConditionExpression {...minimal} />);
    expect(conditionExpression).toMatchSnapshot();
  });
});
