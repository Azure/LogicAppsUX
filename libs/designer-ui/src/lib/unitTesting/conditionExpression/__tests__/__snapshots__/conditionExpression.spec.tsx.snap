// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ui/unitTesting/conditionExpression/index > should render without assertions items and add button. 1`] = `
<React.Fragment>
  <div
    id="condition-expression-condition-expression-editor-id"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onMouseUp={[Function]}
  >
    <ExpressionEditor
      currentHeight={100}
      dragDistance={0}
      editorRef={
        {
          "current": null,
        }
      }
      hideUTFExpressions={false}
      initialValue="@equals(1, 1)"
      isDragging={false}
      isReadOnly={false}
      onBlur={[Function]}
      onContentChanged={[Function]}
      onFocus={[Function]}
      setCurrentHeight={[Function]}
      setExpressionEditorError={[Function]}
      setIsDragging={[Function]}
    />
  </div>
</React.Fragment>
`;
